src/status_im/contexts/wallet/events.cljs

```diff
@@ -328,26 +328,54 @@
+(rf/reg-event-fx
+ :wallet/ens-not-found
+ (fn [{:keys [db]} _]
+   {:db (-> db
+            (assoc-in [:wallet :ui :add-address-to-watch :activity-state] :invalid-ens)
+            (assoc-in [:wallet :ui :add-address-to-watch :validated-address] nil))}))
+
+(rf/reg-event-fx
+ :wallet/store-valid-address-activity
+ (fn [{:keys [db]} [address {:keys [hasActivity]}]]
+   (let [registered-addresses (-> db :wallet :accounts keys set)
+         already-registered?  (registered-addresses address)]
+     (if already-registered?
+       {:db (-> db
+                (assoc-in [:wallet :ui :add-address-to-watch :activity-state]
+                          :address-already-registered)
+                (assoc-in [:wallet :ui :add-address-to-watch :validated-address] nil))}
+       (let [state (if hasActivity :has-activity :no-activity)]
+         {:db (-> db
+                  (assoc-in [:wallet :ui :add-address-to-watch :activity-state] state)
+                  (assoc-in [:wallet :ui :add-address-to-watch :validated-address] address))})))))
 
-(rf/reg-event-fx :wallet/clear-address-activity-check
+(rf/reg-event-fx
+ :wallet/clear-address-activity
  (fn [{:keys [db]}]
-   {:db (update-in db [:wallet :ui] dissoc :watch-address-activity-state)}))
+   {:db (update-in db [:wallet :ui] dissoc :add-address-to-watch)}))
 
-(rf/reg-event-fx :wallet/get-address-details
- (fn [{:keys [db]} [address]]
-   {:db (assoc-in db [:wallet :ui :watch-address-activity-state] :scanning)
-    :fx [[:json-rpc/call
-          [{:method     "wallet_getAddressDetails"
-            :params     [(chain/chain-id db) address]
-            :on-success [:wallet/get-address-details-success]
-            :on-error   #(log/info "failed to get address details"
-                                   {:error %
-                                    :event :wallet/get-address-details})}]]]}))
+(rf/reg-event-fx
+ :wallet/get-address-details
+ (fn [{:keys [db]} [address-or-ens]]
+   (let [request-params [(chain/chain-id db) address-or-ens]
+         ens?           (string/includes? address-or-ens ".")]
+     {:db (-> db
+              (assoc-in [:wallet :ui :add-address-to-watch :activity-state] :scanning)
+              (assoc-in [:wallet :ui :add-address-to-watch :validated-address] nil))
+      :fx [(if ens?
+             [:json-rpc/call
+              [{:method     "ens_addressOf"
+                :params     request-params
+                :on-success [:wallet/get-address-details]
+                :on-error   [:wallet/ens-not-found]}]]
+             [:json-rpc/call
+              [{:method     "wallet_getAddressDetails"
+                :params     request-params
+                :on-success [:wallet/store-valid-address-activity address-or-ens]
+                :on-error   #(log/info "failed to get address details"
+                                       {:error %
+                                        :event :wallet/get-address-details})}]])]})))
```

Comment:
Part of the main solution, I'm calling `"ens_addressOf"` to first resolve the given `ens` to an address